Numerical model for calculating airspeed from venturi effect airspeed sensor (TFSLOT)

roman-dvorak opened the pull request:
We developed airspeed sensor [TFSLOT01](https://github.com/ThunderFly-aerospace/TFSLOT01), that is based on [Venturi effect](https://en.wikipedia.org/wiki/Venturi_effect). This has a number of advantages over the pitot tube, which is often used. Especially for slow-flying machines and aircraft models. 

TFSLOT is equipped with a differential pressure sensor SDP3x. And compared to pitot's tube, it has the opposite polarity. 

The default profile is designed so that the model corresponds as closely as possible to the equation from the pitot tube. Therefore, it is primarily a change of polarity.

In the future, we expect to expand to other profiles (which may be even more sensitive to low speeds) and then we will add certain parameters to this profile that will affect this conversion.

Now I would like to measure the minimum support for TFSLOT. A documentation page for TFSLOT will follow.

### How it is solved? 
Another computational model is added to PX4 (based on the model for the pitot tube). The selection of the model is by a parameter, which is extended by another option. This profile assumes the reverse polarity of the measured pressure. (negative values)

roman-dvorak committed:
add calculation of airspeed profile for TFSLOT, selectable by parameter

roman-dvorak commented:
I just tested the functionality of this adjustment with the [TFSLOT](https://github.com/ThunderFly-aerospace/TFSLOT01) and [TFASPDIMU01B](https://github.com/ThunderFly-aerospace/TFASPDIMU02) sensors (Snserion SDP3x)

#### Testing steps:

1. Compile code
2. Upload it to CUAV NANO
3. Set the correct conversion model
4. Calibrate it 
5. And I checked that the airspeed was measuring reasonable values 

#### Complications:
Calibration is a bit more difficult because the PX4 does not support negative calibration. So it is necessary to perform the calibration by blowing backwards (I will describe the procedure exactly in the documentation).

roman-dvorak commented:
@dagar  please can you do a review or accept it?

dagar commented on src/modules/sensors/sensor_params.c:
```diff
@@ -43,10 +43,13 @@
  * @value 0 Model with Pitot
  * @value 1 Model without Pitot (1.5 mm tubes)
  * @value 2 Tube Pressure Drop
+ * @value 3 Venturi effect airspeed sensor
```
```suggestion
 * @value 3 Venturi effect airspeed sensor
```

Should this be more specific?

roman-dvorak replied:
I slightly expanded the description of the parameter. What exactly would you imagine here?

dagar commented on src/modules/sensors/sensor_params.c:
```diff
@@ -43,10 +43,13 @@
  * 		CAL_AIR_TUBED_MM: Diameter in mm of the pitot and tubes, must have the same diameter.
  * 		CAL_AIR_TUBELEN: Length of the tubes connecting the pitot to the sensor and the static + dynamic port length of the pitot.
+ * Venturi effect sensor
+ *    Supports TFSLOT airspeed sensor
```
```suggestion
 *    Supports TFSLOT airspeed sensor
```

Is there a more google-able name than TFSLOT?

roman-dvorak replied:
We have prepared a documentation page that should improve gogleability of TFSLOT https://github.com/PX4/PX4-user_guide/blob/2a06420dd35cdeeacef266a8fe1d8e16c65e4cd5/en/sensor/airspeed_tfslot.md

Then the README expansion in the TFSLOT repository should also help.

roman-dvorak replied:
We have prepared a documentation page that should improve gogleability of TFSLOT https://github.com/PX4/PX4-user_guide/blob/2a06420dd35cdeeacef266a8fe1d8e16c65e4cd5/en/sensor/airspeed_tfslot.md

Then the README expansion in the TFSLOT repository should also help.

roman-dvorak replied:
The basic documentation for TFSLOT is published. After receiving this PR, I will expand it with settings with this profile. Is it possible to merge it?

kaklik replied:
Hi, I have found [this TFSLOT documentation page](https://docs.px4.io/master/en/sensor/airspeed_tfslot.html) by using PX4 TFSLOT keywords now.  Is it acceptable to resolve that issue?

ryanjAA commented:
This is pretty cool. Looking forward to trying one.

kaklik commented:
Hello @dagar and @bresch,  the conversations in this pull request seem to be resolved for me. Is it possible to merge the changes, please?

roman-dvorak commented:
@dagar, please, can you check it?

roman-dvorak committed:
Update sensor_params.c
Update sensor_params.c
Update airspeed.cpp
Update airspeed.h
Update sensor_params.c, sync submodule

roman-dvorak committed:
Add new airspeed equation for TFSLOT (or Venturi based aspd sensors)
